zerver/models/streams.py

```diff
@@ -186,6 +186,9 @@ def is_history_realm_public(self) -> bool:
     def is_history_public_to_subscribers(self) -> bool:
         return self.history_public_to_subscribers
 
+    def is_support_stream(self) -> bool:
+        return self.stream_topic_access_group.named_user_group.name != SystemGroups.EVERYONE
+
```

Comment:
Accessing `stream_topic_access_group.named_user_group.name` is leading to an inflated number of database queries. Unfortunately, I can't find a way around.

Making `stream_topic_access_group` be foreign key on `NamedUserGroup` will reduce the db query by 1.

Reply:
You can use `select_related` to fetch these fields in the query which fetches the stream.

Reply:
Done.